packages/core/src/integrations/mcp-server/resultExtraction.ts

```diff
@@ -0,0 +1,123 @@
+/**
+ * Result extraction functions for MCP server instrumentation
+ *
+ * Handles extraction of attributes from tool and prompt execution results.
+ */
+
+import {
+  MCP_PROMPT_RESULT_DESCRIPTION_ATTRIBUTE,
+  MCP_PROMPT_RESULT_MESSAGE_COUNT_ATTRIBUTE,
+  MCP_TOOL_RESULT_CONTENT_COUNT_ATTRIBUTE,
+  MCP_TOOL_RESULT_IS_ERROR_ATTRIBUTE,
+} from './attributes';
+
+/**
+ * Build attributes for tool result content items
+ * @param content - Array of content items from tool result
+ * @returns Attributes extracted from each content item including type, text, mime type, URI, and resource info
+ */
+function buildAllContentItemAttributes(content: unknown[]): Record<string, string | number> {
+  const attributes: Record<string, string | number> = {
+    [MCP_TOOL_RESULT_CONTENT_COUNT_ATTRIBUTE]: content.length,
+  };
+
+  for (const [i, item] of content.entries()) {
+    if (typeof item !== 'object' || item === null) continue;
+
+    const contentItem = item as Record<string, unknown>;
+    const prefix = content.length === 1 ? 'mcp.tool.result' : `mcp.tool.result.${i}`;
+
+    const safeSet = (key: string, value: unknown): void => {
+      if (typeof value === 'string') attributes[`${prefix}.${key}`] = value;
+    };
+
+    safeSet('content_type', contentItem.type);
+    safeSet('mime_type', contentItem.mimeType);
+    safeSet('uri', contentItem.uri);
+    safeSet('name', contentItem.name);
+
+    if (typeof contentItem.text === 'string') {
+      const text = contentItem.text;
```

Comment:
l: if we're going to do this, I would prefer if we just destructure it above, and use it in here.

```
const { text } = contentItem;
```

Reply:
fixed, no need for that. thank for being polite about this 😆